mvlearn/semi_supervised/ctregression.py

```diff
@@ -0,0 +1,580 @@
+
+    def _check_params(self):
+        r"""
+        Checks that cotraining parameters are valid. Throws AttributeError
+        if estimators are invalid. Throws ValueError if any other parameters
+        are not valid. The checks performed are:
+            - estimator1 and estimator2 are KNeigborsRegressor
+            - neighbors_size is positive
+            - subsample_size is positive
+            - n_iter is positive
+            - selected_size is positive
+        """
+
+        # The estimator must be KNeighborsRegressor
+        to_be_matched = "KNeighborsRegressor"
+
+        # Taking the str of a class returns the class name
+        # along with other parameters
+        string1 = str(self.estimator1)
+
+        # slicing the list to get the name of the estimator
+        string1 = string1[: len(to_be_matched)]
+
+        # Taking the str of a class returns the class name
+        # along with other parameters
+        string2 = str(self.estimator2)
+
+        # slicing the list to get the name of the estimator
+        string2 = string2[: len(to_be_matched)]
+
+        if string1 != to_be_matched or string2 != to_be_matched:
+            raise AttributeError(
+                "Both the estimator need to be KNeighborsRegressor")
+
+        if self.neighbors_size <= 0:
+            raise ValueError("neighbors size must be positive")
+
+        if self.subsample_size <= 0:
+            raise ValueError("subsample size must be positive")
+
+        if self.n_iter <= 0:
+            raise ValueError("number of iterations must be positive")
+
+        if self.selected_size <= 0:
+            raise ValueError("selected size must be positive")
+
+    def fit(self, Xs, y):
+        r"""
+        Fit the regressor object to the data in Xs, y.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            A list of the different views of data to train on.
+
+        y : array, shape (n_samples,)
+            The labels of the training data. Unlabeled_pool examples should
+            have label np.nan.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+
+        # check whether Xs contain NaN and both Xs and y
+        # are consistent woth each other
+        check_Xs_y_nan_allowed(
+            Xs, y, multiview=True, enforce_views=self.n_views)
+
+        # Xs contain two views
+        view1 = Xs[0]
+        view2 = Xs[1]
+
+        # Storing the indexes of the unlabelled samples
+        U = [i[0] for i in enumerate(y) if np.isnan(i[1])]
+
+        # Storing the indexes of the labelled samples
+        L = [i[0] for i in enumerate(y) if not np.isnan(i[1])]
+
+        X1 = view1
+        X2 = view2
+
+        # making two true labels for each view
+        # So that we can make changes to it without altering original labels
+        y1 = y
+        y2 = y
```

Comment:
I think you need to set y1 = pd.Series(y) or somehow turn it into pandas, because mvlearn API accepts a numpy array